tasks.py

```diff
@@ -268,6 +262,56 @@ def build(_c, debug=False):
         image.tag(constants.IMAGE, tag=latest_tag)
 
 
+@task
+def sbom(_c, debug=False):
+    """Generate an SBOM"""
+    if debug:
+        getLogger().setLevel("DEBUG")
+
+    for variant in constants.VARIANTS:
+        versioned_tag = CONTEXT[variant]["buildargs"]["VERSION"]
+        image_and_tag = f"{constants.IMAGE}:{versioned_tag}"
+
+        try:
+            artifact_labels = utils.get_artifact_labels(variant=variant)
+
+            for iteration, label in enumerate(artifact_labels):
+                if iteration > 0:
+                    prior_label = artifact_labels[iteration - 1]
+                    prior_file_name = f"sbom.{prior_label}.json"
+                file_name = f"sbom.{label}.json"
+
+                if Path(file_name).is_file() and Path(file_name).stat().st_size > 0:
+                    LOG.info(f"Skipping {file_name} because it already exists...")
+                    continue
+
+                if iteration > 0:
+                    LOG.info(
+                        f"Copying {prior_file_name} into {file_name} since they are the same..."
+                    )
+                    shutil.copy(prior_file_name, file_name)
+                    continue
+
+                LOG.info(f"Generating {file_name} from {image_and_tag}...")
+                subprocess.run(
+                    [
+                        "syft",
+                        f"docker:{image_and_tag}",
```

Comment:
I tested to confirm that this pulls from the docker daemon, so it works well if you reference the `latest*` tags, assuming you built the images immediately prior.

Reply:
Noted. That's usually how I do a PR review.